packages/eas-cli/src/commands/worker/deploy.ts

```diff
@@ -0,0 +1,195 @@
+    .stat(directoryPath)
+    .then(stat => stat.isDirectory())
+    .catch(() => false);
+
+export default class WorkerDeploy extends EasCommand {
+  static override description = 'deploy an Expo web build';
+  static override aliases = ['deploy'];
+
+  // TODO(@kitten): Keep command hidden until worker deployments are live
+  static override hidden = true;
+  static override state = 'beta';
+
+  static override flags = {
+    // TODO(@kitten): Allow deployment identifier to be specified
+  };
+
+  static override contextDefinition = {
+    ...this.ContextOptions.ProjectConfig,
+    ...this.ContextOptions.ProjectDir,
+    ...this.ContextOptions.LoggedIn,
+  };
+
+  async runAsync(): Promise<void> {
+    Log.warn('EAS Worker Deployments are in beta and subject to breaking changes.');
+
+    const {
+      privateProjectConfig: { projectId, exp },
+      loggedIn: { graphqlClient },
+      projectDir,
+    } = await this.getContextAsync(WorkerDeploy, {
+      nonInteractive: true,
+    });
+
+    const distPath = path.resolve(projectDir, 'dist');
+    const distClientPath = path.resolve(distPath, 'client');
+    const distServerPath = path.resolve(distPath, 'server');
+    if (!(await isDirectory(distPath))) {
+      throw new Error(
+        `No "dist/" folder found at ${distPath}. Prepare your project for deployment with "npx expo export"`
+      );
+    } else if (!(await isDirectory(distClientPath))) {
+      throw new Error(
+        `No "dist/client/" folder found at ${distClientPath}. Ensure the app.json key "expo.web.output" is set to "server"`
+      );
+    } else if (!(await isDirectory(distServerPath))) {
+      throw new Error(
+        `No "dist/server/" folder found in ${distServerPath}. Ensure the app.json key "expo.web.output" is set to "server"`
+      );
+    }
+
+    async function* emitWorkerTarballAsync(
+      assetMap: WorkerAssets.AssetMap
+    ): AsyncGenerator<WorkerAssets.FileEntry> {
+      yield ['assets.json', JSON.stringify(assetMap)];
+
+      // TODO: Create manifest from user configuration
+      const manifest = { env: {} };
+      yield ['manifest.json', JSON.stringify(manifest)];
+
+      const workerFiles = WorkerAssets.listWorkerFiles(distServerPath);
+      for await (const workerFile of workerFiles) {
+        yield [`server/${workerFile.normalizedPath}`, workerFile.data];
+      }
+    }
+
+    async function uploadTarballAsync(tarPath: string): Promise<any> {
+      const uploadUrl = await getSignedDeploymentUrlAsync(graphqlClient, exp, {
+        appId: projectId,
+      });
+
+      const { response } = await uploadAsync({
+        url: uploadUrl,
+        filePath: tarPath,
+        compress: false,
+        headers: {
+          accept: 'application/json',
+        },
+      });
+      if (response.status === 413) {
+        throw new Error(
+          'Upload failed! (Payload too large)\n' +
+            `The gzipped files in "dist/server/" (at: ${distServerPath}) exceed the maximum file size.`
```

Comment:
What's the maximum file size and/or what can the user do to get "unblocked"? Does the server return a special header with the maximum size?

Reply:
There's no path right now to unblock users, but it's 10MB gzipped. We'll have to explore what we can do in these cases to let users avoid this. One solution is to try to leave out sourcemaps and issue a warning to people, but if 10MB is still exceeded afterwards, we'll have to try some other solutions.

For server components, we may have to split up the server workers if we get close to reaching these limits.